lib/yard/parser/ruby/ruby_parser.rb

```diff
@@ -390,16 +390,12 @@ def on_aref_field(*args)
 
         def on_array(other)
           node = AstNode.node_class_for(:array).new(:array, [other])
-          map = @map[MAPPINGS[node.type]]
+          map = @map[MAPPINGS[node.type]] if other.nil? || other.type == :list
           if map && !map.empty?
             lstart, sstart = *map.pop
             node.source_range = Range.new(sstart, @ns_charno - 1)
             node.line_range = Range.new(lstart, lineno)
           else
-            sstart = other.source_range.begin
-            lstart = other.line_range.begin
-            node.source_range = Range.new(sstart, @ns_charno - 1)
-            node.line_range = Range.new(lstart, lineno)
```

Comment:
Unless I'm missing something, this is dead code.  Let me know if I'm wrong.

Reply:
This is not dead code per se. This is meant to future-proof the implementation for new node types which happens fairly often.

Reply:
Gotcha.  Just seemed strange those two assignments were immediately overridden.  I'll revert that, but definitely curious to hear what I'm misunderstanding there.

Reply:
Reverted